### PR TITLE
Updated view of authentication webpages. Allowed users to register and login

### PR DESCRIPTION
**Pull request by speters0816**
Features:
- Changed login and register webpage to ask for email instead of username
- Moved register and login functions to `auth.py` sub-package to better organize authentication part of application.
- Login and Register now located at `/auth` prefix. E.g. `/auth/login`
- Register webpage
  - Creates hash of password and adds both password hash and user "email" to sqlite3 database.
  - Inhibits user from registering if no values are given for either password or email.
  - Redirects to login once registered
  - Updated layout and look of register to center input boxes and show dark grey theme
- Login webpage
  - Added functionality that checks if values entered are not empty. 
  - Looks for user entered email and hash of password in database. If a match is found, redirects to index page. 
  - Saves user id in session (cookie) after successful login that can be used for database look ups in other functions
  - Updated layout and look of register to center input boxes and show dark grey theme
 - Added missing import modules to __init__.py to allow for logout and saving of display name in index function.

### CODE REVIEW

File: flask-discord/flaskd/templates/login_1.html [hunk 1]
```diff
@@ -1,18 +1,56 @@
...
+table.center {
+  margin-left: auto; 
+  margin-right: auto;
+  
+}
+div {
+  border: 2px solid white;
+  background-color: #6E6D6D;
+  padding-top: 10px;
+  padding-right: 10px;
```
Comment (spyralspy01) [inline flask-discord/flaskd/templates/login_1.html:20]: The login page looks well made.

**Review by spyralspy01 — APPROVED** [review]
Excellent code. And it works great.

File: flask-discord/flaskd/auth.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            except db.IntegrityError:
+                error = "Email {} is already registered.".format(email)
+
+            else:
+                return redirect(url_for("auth.login"))
+
+        flash(error)
+
+    return render_template("register.html")
+
```
Comment (hemanth5143) [inline flask-discord/flaskd/auth.py:68]: The login and  register function codes are written in such a way that they are quite  understanding to  novice coders.

File: flask-discord/flaskd/schema.sql [hunk 3]
```diff
@@ -4,7 +4,8 @@ DROP TABLE IF EXISTS siteViews;
 
 CREATE TABLE user (
 	id INTEGER PRIMARY KEY AUTOINCREMENT,
-	username TEXT UNIQUE NOT NULL,
+	email TEXT UNIQUE NOT NULL,
+	username TEXT NOT NULL DEFAULT "myspace Tom" CHECK( length(username) <= 32),
 	password TEXT NOT NULL
 );
```
Comment (hemanth5143) [inline flask-discord/flaskd/schema.sql:11]: The database code is so good as it maintained proper indentation style and easy to read and understand.

File: flask-discord/flaskd/templates/register.html [hunk 4]
```diff
@@ -1,25 +1,61 @@
...
-    <label for="Password" class="form-label">Password</label>
+    <label  for="Password" class="form-label">Password</label>
     <input type="password" class="form-control" id="Password" name="password">
   </div>
   <button type="submit" class="btn btn-primary">Register</button>
 </form>
 </div>
+</td>
+</tr>
+</table>
```
Comment (hemanth5143) [inline flask-discord/flaskd/templates/register.html:59]: The components placement display on the application home page is well balanced.

**Review by hemanth5143 — APPROVED** [review]

**Review by thegoldenmule — APPROVED** [review]
Looks great! Excellent notes.

**Pull request merged**